File: src/tests/Behavioral/BlankImportSideEffects/pnglike/pnglike.go
// Package pnglike plays the "image/png" role: it exists to be BLANK-imported. Nothing outside it
// ever names one of its identifiers, so the only reason its converted assembly must be loaded is
// this init — exactly what Go guarantees and what .NET's first-use module-initializer rule does not.
package pnglike

import "BlankImportSideEffects/registry"

func init() {
	registry.Register("pnglike", "pnglike decoder")
}

File: src/tests/Behavioral/BlankImportSideEffects/pnglike/pnglike.cs
namespace go.BlankImportSideEffects;

using registry = go.BlankImportSideEffects.registry_package;
using go.BlankImportSideEffects;

partial class pnglike_package {

[GoInit] internal static void init() {
    registry.Register("pnglike"u8, "pnglike decoder"u8);
}

} // end pnglike_package
